components/Ink/src/private/MDCSimpleInkLayer.m

```diff
@@ -0,0 +1,162 @@
+  self.opacity = 0;
+  self.position = point;
+  [self startAnim:point];
+}
+
+- (void)end {
+  [self endAnim];
+}
+
+- (void)startAnim:(CGPoint)point {
+  self.startAnimActive = YES;
+
+  CAKeyframeAnimation *scaleAnim = [[CAKeyframeAnimation alloc] init];
+  [scaleAnim setKeyPath:@"transform.scale"];
+  scaleAnim.keyTimes = @[ @0, @1.0f ];
+  scaleAnim.values = @[ @0.6f, @1.0f ];
+  scaleAnim.duration = 0.333f;
+  scaleAnim.beginTime = 0.083f;
+  scaleAnim.timingFunction = [[CAMediaTimingFunction alloc] initWithControlPoints:0.4f:0:0.2f:1.f];
+  scaleAnim.fillMode = kCAFillModeForwards;
+  scaleAnim.removedOnCompletion = NO;
+
+  UIBezierPath *centerPath = [UIBezierPath bezierPath];
+  CGPoint startPoint = point;
+  CGPoint endPoint = CGPointMake(self.frame.size.width / 2, self.frame.size.height / 2);
+  [centerPath moveToPoint:startPoint];
+  [centerPath addLineToPoint:endPoint];
+  [centerPath closePath];
+
+  CAKeyframeAnimation *positionAnim = [[CAKeyframeAnimation alloc] init];
+  [positionAnim setKeyPath:@"position"];
+  positionAnim.path = centerPath.CGPath;
+  positionAnim.keyTimes = @[ @0, @1.0f ];
+  positionAnim.values = @[ @0, @1.0f ];
+  positionAnim.duration = 0.333f;
+  positionAnim.beginTime = 0.083f;
+  positionAnim.timingFunction =
+      [[CAMediaTimingFunction alloc] initWithControlPoints:0.4f:0:0.2f:1.f];
+  positionAnim.fillMode = kCAFillModeForwards;
+  positionAnim.removedOnCompletion = NO;
+
+  CAKeyframeAnimation *fadeInAnim = [[CAKeyframeAnimation alloc] init];
+  [fadeInAnim setKeyPath:@"opacity"];
+  fadeInAnim.keyTimes = @[ @0, @1.0f ];
+  fadeInAnim.values = @[ @0, @1.0f ];
+  fadeInAnim.duration = 0.083f;
+  fadeInAnim.beginTime = 0.083f;
+  fadeInAnim.timingFunction = [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionLinear];
+  fadeInAnim.fillMode = kCAFillModeForwards;
+  fadeInAnim.removedOnCompletion = NO;
+
+  CAKeyframeAnimation *fadeHalfAnim = [[CAKeyframeAnimation alloc] init];
+  [fadeHalfAnim setKeyPath:@"opacity"];
+  fadeHalfAnim.keyTimes = @[ @0, @1.0f ];
+  fadeHalfAnim.values = @[ @1.0f, @0.5f ];
+  fadeHalfAnim.duration = 0.25f;
+  fadeHalfAnim.beginTime = 0.167f;
+  fadeHalfAnim.timingFunction =
+      [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionLinear];
+  fadeHalfAnim.fillMode = kCAFillModeForwards;
+  fadeHalfAnim.removedOnCompletion = NO;
+
+  [CATransaction begin];
+  CAAnimationGroup *animGroup = [[CAAnimationGroup alloc] init];
+  animGroup.animations = @[ scaleAnim, positionAnim, fadeInAnim, fadeHalfAnim ];
+  animGroup.duration = 0.5f;
+  animGroup.fillMode = kCAFillModeForwards;
+  animGroup.removedOnCompletion = NO;
+  [CATransaction setCompletionBlock:^{
+    self.startAnimActive = NO;
+  }];
+  [self addAnimation:animGroup forKey:nil];
+  [CATransaction commit];
+}
+
+- (void)endAnim {
+  if (self.startAnimActive) {
+    self.endAnimDelay = 0.25f;
+  }
+  NSNumber *currOpacityNumber = [self.presentationLayer valueForKeyPath:@"opacity"];
```

Comment:
Can we not invoke self.presentationLayer.opacity?

Reply:
self.presentationLayer.opacity appears to be ok.